src/main/java/com/amazon/aocagent/tasks/ECSClean.java

```diff
@@ -0,0 +1,70 @@
+package com.amazon.aocagent.tasks;
+
+import com.amazon.aocagent.enums.GenericConstants;
+import com.amazon.aocagent.models.Context;
+import com.amazon.aocagent.services.EC2Service;
+import com.amazon.aocagent.services.ECSService;
+import com.amazonaws.services.ec2.model.Instance;
+import lombok.extern.log4j.Log4j2;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+@Log4j2
+public class ECSClean implements ITask {
+  private EC2Service ec2Service;
+  private ECSService ecsService;
+
+  @Override
+  public void init(Context context) throws Exception {
+    this.ec2Service = new EC2Service(context.getStack().getTestingRegion());
+    this.ecsService = new ECSService(context.getStack().getTestingRegion());
+  }
+
+  @Override
+  public void execute() throws Exception {
+
+    cleanTasks();
+    cleanContainerInstances();
+    // static cluster name, might don't need to be cleaned for now
+    //        cleanCluster();
+    cleanTaskDefinitions();
+  }
+
+  private void cleanTaskDefinitions() {
+    ecsService.cleanTaskDefinitions(GenericConstants.AOC_PREFIX.getVal());
+  }
+
+  private void cleanCluster() {
+    ecsService.cleanCluster(GenericConstants.ECS_SIDECAR_CLUSTER.getVal());
```

Comment:
i think you need list all the clusters with tag, and check the created time one by one. there's no created_time in the Cluster object, so one possible way is to append a timestamp to the cluster name